### PR TITLE
Add tests for Redux sample code

### LINKED ISSUE #15: Tests for Redux sample code
## Motivation
Just as important as having code samples for functionality in Redux
is having test code samples for that functionality. Since one of the
core strengths of Redux is its tendency to produce easily testable
code, it would be good to have some examples
from the start. Testing redux-observable code looks slightly tricker,
so those examples will be all the more helpful to have.

## Plan of action
- [x] Unit tests for all actions
  - Assert that, for each action, the state changes in the expected way
  - Heavily leaning towards [this approach](https://github.com/reactjs/redux/issues/2179), wherein you test the entire store; i.e. dispatch actions and use selectors to assert the expected state change
- [x] Unit tests for all epics
  - This is a bit of a question mark; I need to do more research into how to best set this up.

*The rest were split off into #22 and #23*
- ~Unit tests for components~
  - ~Where appropriate. Some components don't do anything yet.~
- ~Functional tests with mock API~
  - ~Use something like enzyme to render an entire screen, simulate user interactions, and assert things about what happens on the screen.~

### PR DESCRIPTION
**Pull request by brettdh**
~Adding commits as I check off the items in #15.~
Split off #22 and #23 from #15. Moving away from WIP PRs (i.e. those that are "ready for review but will be added to and then need to be re-reviewed") in favor of self-contained PRs that usually don't need to be re-reviewed - or, in the case of "I want some feedback on code that's not ready to be merged yet", a clear label that indicates the purpose of the PR.

Basically I realized that, once someone approves a PR, there's no way to see whether they've also read and approved any commits that land after their approval. (There is a setting on the repo that resets approvals on push, but I don't want to make reviewers come back just for e.g. typo fixes.)

Closes #15.

### CODE REVIEW

**Review by wittjosiah — APPROVED** [review]
🙌 tests! LGTM :shipit:
